### ISSUE TITLE
ExDates Don't Seem to Work

### ISSUE THREAD

**Issue by Treebasher** [dmfs/lib-recur#56]
**TL;DR:** Exception Dates added via `RecurrenceSet.addExceptions(...)` don't seem to actually prevent iteration to recurrences that would normally occur on added dates.

For example, assume I have a `RecurrenceSet` object named `ruleSet`. I first call `ruleSet.addInstances(...)` on my set object to add a simple Recurrence Rule such as `FREQ=DAILY;COUNT=5`. I then call something like `ruleSet.addExceptions(new RecurrenceList("20190125", null))` to add a single Exception Date (`January 25, 2019`) to my `ruleSet`.

If I then use `ruleSet.iterator(...)` to get an iterator, starting on `January 24, 2019` (at whatever time the Event starts at), and proceed to iterate over the recurrences it provides, I will still end up being handed a timestamp corresponding to `January 25, 2019` (at whatever time the Event starts at) in the time zone I provided.

This seems to be broken functionality. Further, I looked through the tests for this library and couldn't find any that actually test ExDates. I'm wondering if I am using this feature of the library wrong, or if this is a bug?

**Comment by dmfs** [comment]
Depends, if the event has a time, the EXDATE must have exactly the same time. It's not like you just specify a date and all events on that date are skipped. Check out this discussion regarding the question whether this is correct behaviour (for an RFC 5545 based implementation) or not: https://bugs.kde.org/show_bug.cgi?id=169336

Is the event correctly excluded when your start date doesn't have a time or when your EXDATE specifies the same time as the start event?

**Comment by Treebasher** [comment]
This makes sense. The use case that hit it home for me is that, if a meeting is scheduled for people in multiple time zones, and will end up occurring on one day in one time zone of interest, but on another day in another time zone of interest (same instant), creating a floating exception date may only cause the meeting to "not occur" in one of the multiple time zones where people are supposedly attending said meeting.

Specifying the exact timestamp of the occurrence to exclude does seem to work properly. Thank you — I'm going to close this issue.
